Check DoNotEvict after filtering evictable pods to ensure termination can complete. (#1294)
aws/karpenter: website/content/en/preview/tasks/deprovisioning.md

### Pod set to do-not-evict

> If a pod exists with the annotation `karpenter.sh/do-not-evict` on a node, and a request is made to delete the node, Karpenter will not drain any pods from that node or otherwise try to delete the node. This annotation does not apply to static pods, pods that tolerate `NoSchedule`, or pods terminating past their graceful termination period.

Reviewer: nit in wording: 

```suggestion
If a pod exists with the annotation `karpenter.sh/do-not-evict` on a node, and a request is made to delete the node, Karpenter will not drain any pods from that node or otherwise try to delete the node. This annotation will have no effect for static pods, pods that tolerate `NoSchedule`, or pods terminating past their graceful termination period.
```

This is useful for pods that you want to run from start to finish without interruption.
Examples might include a real-time, interactive game that you don't want to interrupt or a long batch job (such as you might have with machine learning) that would need to start over if it were interrupted.

> If you want to terminate node with a `do-not-evict` pod, you can simply remove the annotation and the deprovisioning process will continue.

Reviewer: s/node with/a node with or s/node with/nodes with

Author: Oh nice catch!
